### ISSUE TITLE
An exception occurred while executing 'UPDATE `oc_deck_cards` SET 

### ISSUE THREAD

**Issue by Githopp192** [nextcloud/deck#321]
**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)

**Where did you install Nextcloud from:**

**Signing status:**

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```

**List of activated apps:**
### Steps to reproduce
1### Steps to reproduce
1.
2.
3.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### General server configuration

**Operating system:** FreeBSD Nextcloud 11.0-RELEASE FreeBSD 11.0-STABLE #0 r321665+25fe8ba8d06(freenas/11.0-stable): Mon Sep 25 06:24:11 UTC 2017     root@gauntlet:/freenas-11-releng/freenas/_BE/objs/freenas-11-releng/freenas/_BE/os/sys/FreeNAS.amd64 amd64

**Web server:** Apache/2.4.27 (FreeBSD) OpenSSL/1.0.2j-freebsd PHP/7.0.21 (apache2handler)

**Database:** mysql 10.2.7

**PHP version:** 7.0.21
<details>
        <summary>PHP-modules loaded</summary>

```
 - Core
 - date
 - libxml
 - pcre
 - Reflection
 - SPL
 - hash
 - session
 - apache2handler
 - standard
 - apcu
 - bz2
 - ctype
 - curl
 - dom
 - mbstring
 - fileinfo
 - filter
 - gd
 - mysqlnd
 - iconv
 - intl
 - json
 - exif
 - mcrypt
 - openssl
 - PDO
 - posix
 - SimpleXML
 - xml
 - xmlwriter
 - zip
 - zlib
 - memcached
 - pdo_mysql
 - wddx
 - xmlreader
 - xsl
 - Zend OPcache
```
</details>

### Nextcloud configuration

**Nextcloud version:** 12.0.3 - 12.0.3.3

**Updated from an older Nextcloud/ownCloud or fresh install:** YOUR ANSWER HERE

Update from 12.0.2

**Where did you install Nextcloud from:**  YOUR ANSWER HERE

Nextcloud automatic WEB Installer

**Are you using external storage, if yes which one:** Array

NO

(
    [0] => \OC\Files\Storage\Local
    [1] => \OC\Files\Storage\DAV
    [2] => \OCA\Files_External\Lib\Storage\OwnCloud
    [3] => \OCA\Files_External\Lib\Storage\SFTP
    [4] => \OCA\Files_External\Lib\Storage\AmazonS3
    [5] => \OCA\Files_External\Lib\Storage\Dropbox
    [6] => \OCA\Files_External\Lib\Storage\Google
    [7] => \OCA\Files_External\Lib\Storage\Swift
    [8] => \OCA\Files_External\Lib\Storage\SFTP
    [9] => OCA\SharePoint\Storage\Storage
)

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** YOUR ANSWER HERE (LDAP/ActiveDirectory/Webdav/...)

<details>
        <summary>Signing status</summary>

```
[]
```
</details>

<details>
        <summary>Enabled apps</summary>

```
 - activity: 2.5.2
 - announcementcenter: 3.1.0
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.5
 - circles: 0.12.4
 - comments: 1.2.0
 - contacts: 2.0.1
 - dav: 1.3.0
 - deck: 0.2.4
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_clipboard: 0.6.4
 - files_external: 1.3.0
 - files_pdfviewer: 1.1.1
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - impersonate: 1.0.1
 - issuetemplate: 0.2.2
 - keeweb: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.7.3
 - mood: 0.3.2
 - music: 0.5.2
 - nextcloud_announcements: 1.1
 - notes: 2.3.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - ownbackup: 17.5.0
 - passman: 2.1.4
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - quota_warning: 1.1.0
 - ransomware_protection: 1.0.4
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - sharepoint: 1.0.2
 - socialsharing_diaspora: 1.0.1
 - socialsharing_email: 1.0.1
 - socialsharing_facebook: 1.0.1
 - socialsharing_googleplus: 1.0.1
 - socialsharing_twitter: 1.0.1
 - spreed: 2.0.1
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
 - admin_audit
 - apporder
 - appvncsafe
 - audioplayer
 - documents
 - encryption
 - ocdownloader
 - ojsxc
 - onlyoffice
 - richdocuments
 - spreedme
 - user_external
 - user_ldap
```
</details>

<details>
        <summary>Content of config/config.php</summary>

```
{
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/tmp\/redis.sock",
        "port": 0,
        "dbindex": 0,
        "password": "***REMOVED SENSITIVE VALUE***",
        "timeout": 1.5
    },
    "instanceid": "ocnudqtqmhau",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "REMOVED SENSITIVE VALUE",
        "REMOVED SENSITIVE VALUE"
    ],
    "datadirectory": "\/media",
    "overwrite.cli.url": "http:\/\/REMOVED SENSITIVE VALUE",
    "dbtype": "mysql",
    "version": "12.0.3.3",
    "dbname": "nextcloud",
    "dbhost": "localhost:\/tmp\/mysql.sock",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "maintenance": false,
    "theme": "",
    "loglevel": 0,
    "updater.release.channel": "stable",
    "auth.bruteforce.protection.enabled": true,
    "check_for_working_htaccess": true,
    "data-fingerprint": "REMOVED SENSITIVE VALUE",
    "mail_from_address": "NextCloud",
    "mail_smtpmode": "smtp",
    "mail_smtpauthtype": "LOGIN",
    "mail_domain": "REMOVED SENSITIVE VALUE",
    "mail_smtpsecure": "tls",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "REMOVED SENSITIVE VALUE",
    "mail_smtpport": "587",
    "session_lifetime": 7200,
    "session_keepalive": false
}
```
</details>

### Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063

**Operating system:** YOUR ANSWER HERE
Windows 10

### Logs

<details>
        <summary>Web server error log</summary>

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE `oc_deck_cards` SET `description` = ?,`duedate` = ?,`last_modified` = ? WHERE `id` = ?' with params ["TEST 2", "2017-10-12T10:15:58+00:00", 1507458735, 7]: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '2017-10-12T10:15:58+00:00' for column 'duedate' at row 1
/usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php - line 128: Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
/usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Statement.php - line 177: Doctrine\DBAL\DBALException driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'UPDATE `oc_deck...', Array)
/usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 247: Doctrine\DBAL\Statement->execute()
/usr/local/www/nextcloud/apps/deck/lib/Db/DeckMapper.php - line 40: OCP\AppFramework\Db\Mapper->execute('UPDATE `*PREFIX...', Array, NULL, NULL)
/usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 187: OCA\Deck\Db\DeckMapper->execute('UPDATE `*PREFIX...', Array)
/usr/local/www/nextcloud/apps/deck/lib/Db/CardMapper.php - line 50: OCP\AppFramework\Db\Mapper->update(Object(OCA\Deck\Db\Card))
/usr/local/www/nextcloud/apps/deck/lib/Service/CardService.php - line 92: OCA\Deck\Db\CardMapper->update(Object(OCA\Deck\Db\Card))
/usr/local/www/nextcloud/apps/deck/lib/Controller/CardController.php - line 94: OCA\Deck\Service\CardService->update(7, 'Subject 1', 8, 'plain', 999, 'TEST 2', 'Michael', '2017-10-12T10 1...')
[internal function] OCA\Deck\Controller\CardController->update(7, 'Subject 1', 8, 'plain', 999, 'TEST 2', '2017-10-12T10 1...')
/usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 160: call_user_func_array(Array, Array)
/usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 90: OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Deck\Controller\CardController), 'update')
/usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 114: OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Deck\Controller\CardController), 'update')
/usr/local/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OCA\\Deck\\Contro...', 'update', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/usr/local/www/nextcloud/lib/private/Route/Router.php - line 299: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/usr/local/www/nextcloud/lib/base.php - line 1004: OC\Route\Router->match('/apps/deck/card...')
/usr/local/www/nextcloud/index.php - line 48: OC handleRequest()
{main}

```
Insert your webserver log here
```
</details>

<details>
        <summary>Nextcloud log (data/nextcloud.log)</summary>

```
Insert your Nextcloud log here
```
</details>

<details>
        <summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>
c) ...
```

**Comment by Githopp192** [comment]
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE `oc_deck_cards` SET `description` = ?,`duedate` = ?,`last_modified` = ? WHERE `id` = ?' with params ["TEST 2", "2017-10-12T10:15:58+00:00", 1507458735, 7]: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '2017-10-12T10:15:58+00:00' for column 'duedate' at row 1
/usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php - line 128: Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
/usr/local/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Statement.php - line 177: Doctrine\DBAL\DBALException driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOMySql\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'UPDATE `oc_deck...', Array)
/usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 247: Doctrine\DBAL\Statement->execute()
/usr/local/www/nextcloud/apps/deck/lib/Db/DeckMapper.php - line 40: OCP\AppFramework\Db\Mapper->execute('UPDATE `*PREFIX...', Array, NULL, NULL)
/usr/local/www/nextcloud/lib/public/AppFramework/Db/Mapper.php - line 187: OCA\Deck\Db\DeckMapper->execute('UPDATE `*PREFIX...', Array)
/usr/local/www/nextcloud/apps/deck/lib/Db/CardMapper.php - line 50: OCP\AppFramework\Db\Mapper->update(Object(OCA\Deck\Db\Card))
/usr/local/www/nextcloud/apps/deck/lib/Service/CardService.php - line 92: OCA\Deck\Db\CardMapper->update(Object(OCA\Deck\Db\Card))
/usr/local/www/nextcloud/apps/deck/lib/Controller/CardController.php - line 94: OCA\Deck\Service\CardService->update(7, 'Subject 1', 8, 'plain', 999, 'TEST 2', 'Michael', '2017-10-12T10 1...')
[internal function] OCA\Deck\Controller\CardController->update(7, 'Subject 1', 8, 'plain', 999, 'TEST 2', '2017-10-12T10 1...')
/usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 160: call_user_func_array(Array, Array)
/usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 90: OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Deck\Controller\CardController), 'update')
/usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 114: OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Deck\Controller\CardController), 'update')
/usr/local/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OCA\\Deck\\Contro...', 'update', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/usr/local/www/nextcloud/lib/private/Route/Router.php - line 299: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/usr/local/www/nextcloud/lib/base.php - line 1004: OC\Route\Router->match('/apps/deck/card...')
/usr/local/www/nextcloud/index.php - line 48: OC handleRequest()
{main}

**Comment by juliushaertl** [comment]
Fixed with #322 

backported for 0.2.5 in a5b566e
